src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -53,6 +61,126 @@ struct CPU {
+                std::getline(cache_2_file, cache_2_info);
+
+                if (((endpos = cache_1_info.find(',', pos)) != std::string::npos) ||
+                    ((endpos = cache_1_info.find('-', pos)) != std::string::npos)) {
+                    sub_str = cache_1_info.substr(pos, endpos);
+                    core_1 = std::stoi(sub_str);
+                    sub_str = cache_1_info.substr(endpos + 1);
+                    core_2 = std::stoi(sub_str);
+
+                    _phy_cores++;
+                    _p_cores++;
+                    _cpu_mapping[core_1][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_2][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_1][CPU_MAP_LOG_CORE] = _p_cores;
+                    _cpu_mapping[core_2][CPU_MAP_PHY_CORE] = _p_cores;
+
+                } else if ((endpos = cache_2_info.find('-', pos)) != std::string::npos) {
+                    sub_str = cache_2_info.substr(pos, endpos);
+                    core_1 = std::stoi(sub_str);
+                    sub_str = cache_2_info.substr(endpos + 1);
+                    core_2 = std::stoi(sub_str);
+
+                    _e_cores++;
+                    for (int m = core_1; m <= core_2; m++) {
+                        _phy_cores++;
+                        _cpu_mapping[m][CPU_MAP_CORE] = _phy_cores;
+                        _cpu_mapping[m][CPU_MAP_SMALL_CORE] = _e_cores;
+                    }
+
+                } else {
+                    core_1 = std::stoi(cache_1_info);
+
+                    _p_cores++;
+                    _phy_cores++;
+                    _cpu_mapping[core_1][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_2][CPU_MAP_PHY_CORE] = _p_cores;
+                }
+            }
+            return;
+        };
+
+        for (int n = 0; n < _proc; n++) {
+            if (0 == _cpu_mapping[n][CPU_MAP_SOCKET]) {
+                std::ifstream cache_3_file("/sys/devices/system/cpu/cpu" + std::to_string(n) +
+                                           "/cache/index3/shared_cpu_list");
+                std::string cache_3_info;
+                std::getline(cache_3_file, cache_3_info);
+
+                std::string::size_type pos = 0;
+                std::string::size_type endpos = 0;
+                std::string sub_str;
+
+                int core_1;
+                int core_2;
+
+                _node++;
+                while (1) {
+                    if ((endpos = cache_3_info.find('-', pos)) != std::string::npos) {
+                        sub_str = cache_3_info.substr(pos, endpos);
+                        core_1 = std::stoi(sub_str);
+                        sub_str = cache_3_info.substr(endpos + 1);
+                        core_2 = std::stoi(sub_str);
+
+                        for (int m = core_1; m <= core_2; m++) {
+                            _cpu_mapping[m][CPU_MAP_SOCKET] = _node;
+                            updateProcMapping(m);
+                        }
```

Comment:
If there is two sockets `(eg: 0-17,36-53)`, I can't see _node++ when into the second socket.